### ISSUE TITLE
Functioning `dc.and` syntax is throwing TS errors

### ISSUE THREAD

**Issue by jwhitmarsh** [jawj/zapatos#161]
This code works correctly, but throws TS errors, and I can't figure out how to configure it correctly. Is this a user issue, lib issue/limitation, or a TS limitation?

```const existing = await db
        .select(
            "accommodation_option",
            dc.and(
                { applicantId: args.applicantId },
                dc.or({ applicantDecision: dc.isNull }, dc.not({ applicantDecision: "rejected" })),
            ),
        )
        .run(pool);
```

Throwing: 
```
Argument of type '[{ applicantId: number; }, SQLFragment<boolean | null, never>]' is not assignable to parameter of type 'Whereable[] | SQLFragment<any, never>[]'.
  Type '[{ applicantId: number; }, SQLFragment<boolean | null, never>]' is not assignable to type 'SQLFragment<any, never>[]'.
    Type 'SQLFragment<boolean | null, never> | { applicantId: number; }' is not assignable to type 'SQLFragment<any, never>'.
      Type '{ applicantId: number; }' is not assignable to type 'SQLFragment<any, never>'.
        Object literal may only specify known properties, and 'applicantId' does not exist in type 'SQLFragment<any, never>'.
```

And 
```
Argument of type '[{ applicantDecision: SQLFragment<boolean, never>; }, SQLFragment<boolean | null, never>]' is not assignable to parameter of type 'Whereable[] | SQLFragment<any, never>[]'.
  Type '[{ applicantDecision: SQLFragment<boolean, never>; }, SQLFragment<boolean | null, never>]' is not assignable to type 'SQLFragment<any, never>[]'.
    Type 'SQLFragment<boolean | null, never> | { applicantDecision: SQLFragment<boolean, never>; }' is not assignable to type 'SQLFragment<any, never>'.
      Type '{ applicantDecision: db.SQLFragment<boolean, never>; }' is not assignable to type 'SQLFragment<any, never>'.
        Object literal may only specify known properties, and 'applicantDecision' does not exist in type 'SQLFragment<any, never>'.
```

**Comment by jawj** [comment]
Hmm, thanks for raising this. It ought to work since version 6.1, per issue #105. I'll try to take a look soon. Maybe a recent TS release has changed things.

**Comment by OfficerHalf** [comment]
I've run into this same type error; it seems like the `and` condition wants `Whereable[] | SQLFragment[]` when it should want `Array<Whereable | SQLFragment>`. My workaround was to just wrap the whereable in another `and` condition:

```
Db.select(
    'user_role',
    Db.conditions.and(
      Db.conditions.and({user_reservation_id: Db.conditions.isNotNull}),
    Db.conditions.or(
        { record_keeper_id: Db.conditions.isIn(access.recordKeeperIds) },
        { plan_sponsor_id: Db.conditions.isIn(allPlanSponsorIds) }
    ))
)
```

**Comment by jawj** [comment]
Thanks @OfficerHalf — think that diagnosis was just right.

**Comment by jawj** [comment]
Released in 6.2.
